pkg/template/v2/pem.go

```diff
@@ -56,8 +56,45 @@ func filterPEM(pemType, input string) (string, error) {
 	return string(blocks), nil
 }
 
-func pemEncode(thing, kind string) (string, error) {
+func filterCertChain(certType, input string) (string, error) {
+	ordered, err := fetchX509CertChains([]byte(input))
+	if err != nil {
+		return "", err
+	}
+
+	switch certType {
+	case "leaf":
```

Comment:
Please put these values into a constant. Otherwise, this is good to go. :) Well done!